src/lib/optimizer/strategy/exists_reformulation_rule.cpp

```diff
@@ -152,8 +152,9 @@ void ExistsReformulationRule::apply_to(const std::shared_ptr<AbstractLQPNode>& n
   }
 
   // Build the join node and put it into the LQP in the place of the predicate
-  const auto join_mode =
-      exists_expression->exists_expression_type == ExistsExpressionType::Exists ? JoinMode::Semi : JoinMode::Anti;
+  const auto join_mode = exists_expression->exists_expression_type == ExistsExpressionType::Exists
+                             ? JoinMode::Semi
+                             : JoinMode::AntiRetainNulls;
```

Comment:
Same as below, correct to use the `retain` here?

Reply:
NOT IN == JoinMode::AntiDiscardNulls
NOT EXISTS = JoinMode::AntiRetainNulls

Reply:
Argh, I was thinking about the new reformulation rule and mixed that up. You're absolutely right.